packages/codemods/src/upgrade.ts

```diff
@@ -4,11 +4,18 @@ import { Logger } from './modRunner/logger';
 import { Project } from 'ts-morph';
 // Injection point for logger so that it can easily be replaced.
 const logger: Logger = console;
-export function upgrade(options: CommandParserResult) {
+
+interface UpgradeFunctions {
+  getTsConfigs: () => string[];
+  saveSync: (file: { saveSync: () => void }) => void;
+  saveAsync: (project: { save: () => void }) => void;
+}
+
+export function _upgradeTest(options: CommandParserResult, fns: UpgradeFunctions) {
```

Comment:
Might be better to call this something like `_upgradeInternal` since it's not just for tests.